beacon/types/exec_data_test.go

```diff
@@ -0,0 +1,96 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/beacon/engine"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+func sanitize(input []byte) []byte {
+	sanitized := bytes.ReplaceAll(input, []byte(" "), []byte(""))
+	sanitized = bytes.ReplaceAll(sanitized, []byte("\r"), []byte(""))
+	sanitized = bytes.ReplaceAll(sanitized, []byte("\n"), []byte(""))
+	return sanitized
+}
+
+func decodeEncodeJSON(input []byte, val interface{}) error {
+	if err := json.Unmarshal(input, &val); err != nil {
+		// not valid JSON, nothing to do
+		return nil
+	}
+	output, err := json.Marshal(val)
+	if err != nil {
+		return err
+	}
+	// compare without spaces/CR/LF
+	if !bytes.Equal(sanitize(input), sanitize(output)) {
+		return fmt.Errorf("encode-decode is not equal, \ninput : %x\noutput: %x", input, output)
+	}
+	return nil
+}
+
+func FuzzJSONExecData(f *testing.F) {
+	// Prepare and add a ExecutableData object to the fuzzer.
+	execData := engine.ExecutableData{
+		ParentHash:    common.HexToHash("0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"),
+		FeeRecipient:  common.HexToAddress("0x1234567890abcdef1234567890abcdef12345678"),
+		StateRoot:     common.HexToHash("0xabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdef"),
+		ReceiptsRoot:  common.HexToHash("0xabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdef"),
+		LogsBloom:     types.Bloom{}.Bytes(),
+		Random:        common.HexToHash("0xabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdef"),
+		Number:        123456,
+		GasLimit:      8000000,
+		GasUsed:       4000000,
+		Timestamp:     uint64(time.Now().Unix()),
+		ExtraData:     []byte("extra data"),
+		BaseFeePerGas: big.NewInt(1000000000),
+		BlockHash:     common.HexToHash("0xabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdef"),
+		Transactions: [][]byte{
+			[]byte("transaction1"),
+			[]byte("transaction2"),
+		},
+		Withdrawals: []*types.Withdrawal{
+			{
+				Index:     1,
+				Validator: 1,
+				Address:   common.HexToAddress("0x1234567890abcdef1234567890abcdef12345678"),
+				Amount:    uint64(1000000000),
+			},
+		},
+		BlobGasUsed:   new(uint64),
+		ExcessBlobGas: new(uint64),
+		Deposits: []*types.Deposit{
+			{
+				Index:                 1,
+				WithdrawalCredentials: common.HexToHash("0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"),
+				PublicKey:             [48]byte{},
+				Amount:                uint64(1000000000),
+				Signature:             [96]byte{},
+			},
+		},
+		ExecutionWitness: &types.ExecutionWitness{},
+		WithdrawalsRoot:  &types.EmptyWithdrawalsHash,
+	}
+	input, err := json.Marshal(execData)
+	if err != nil {
+		f.Add(input)
+	}
+	f.Fuzz(fuzzJSONExecData)
```

Comment:
I cannot tell how useful this fuzz test is. The way that I would approach a fuzz test is getting a fuzz input value for each field in the struct rather than bytes that represent the JSON itself